### ISSUE TITLE
Add support for an executable format

### ISSUE THREAD

**Issue by xadaemon** [29jm/SnowflakeOS#28]
I'm opening this to look into discussing this and see if:
* it makes sense right now as the project stands?
* if it does choose between ELF or PE;

the main objective with having an executable format is to make adding applications easier, I'd say we go with elf since that's what the tool-chain is putting out.

**Comment by 29jm** [comment]
It'd make a lot of sense! I tried writing an elf (elf it is) parser at some point, but found an easier workaround for what I wanted to do at the time, so I ended up dumping it, barely started. Having an elf loader would fix a few hacks:
+ the kernel can only load programs whose entry point is at address 0x1000
+ executable files must include the entirety of their data, even .bss sections (see the comment in modules/mod.ld)

and it'd allow for cool things later on (dynamic linking?). Not sure how it would all fit together, `proc_run_code` (in proc.c) or even earlier `proc_exec` could make use of it probably. I'll have more time later this week to investigate. Is it something you'd be interested in working on?

**Comment by xadaemon** [comment]
Yes, I'm interested in doing it, however this one is going to take longer than usual because it's larger and, work has resumed so back to writing saas stuff most of the day it is.

-------- Original Message --------

**Comment by 29jm** [comment]
Yeah I get that, similar situation for me basically, this week especially. Anyway that'd be a long awaited feature, and don't hesitate to open a draft pr early!

**Comment by xadaemon** [comment]
@29jm let's do this assign me and I will prob open the PR pretty early on since I might need a lot more input from you this time around (so we can keep a consistent design).

**Comment by 29jm** [comment]
Alright, let's do this!
